### PR TITLE
[tutorials][ML] Add resampling tutorial

### PR DESCRIPTION
**Pull request by JazzyJonah**
# This Pull request adds a tutorial to show the syntax of resampling

## Changes or fixes:
- New tutorial in /tutorials/machine_learning/ml_dataloader_resampling.py
- There currently isn't a resampling tutorial, so this fixes that!

## Checklist:

- [x] tested changes locally
- [x] updated the docs (if necessary)

### CODE REVIEW

**Comment by silverweed** [comment]
Thanks for the contribution. As a general comment, please squash/reorganize your commits to help the reviewers keep track of your changes.

File: tutorials/machine_learning/ml_dataloader_resampling.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+# and then merge them to form a dataset with underrepresented odd numbers.
+def make_df(b1_expr, num_events):
+    return ROOT.RDataFrame(num_events).Define("b1", b1_expr).Define("b2", "(int) b1%2")
+
+
+df_major = make_df("(int) 2 * rdfentry_", 100000)
+df_minor = make_df("(int) 2 * rdfentry_ + 1", 1000)
+
+batch_size = 256
+batches_in_memory = 10  # When eager_loading is enabled, batches_in_memory isn't ever used
```
Comment (siliataider) [inline tutorials/machine_learning/ml_dataloader_resampling.py]: Then let's not set `batches_in_memory` nor pass it! It's an optional keyword argument

File: tutorials/machine_learning/ml_dataloader_resampling.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+
+# First, let's try to create a dataloader without resampling and see how it handles the underrepresented class.
+dl = ROOT.Experimental.ML.RDataLoader(
+    [df_major, df_minor],
+    batch_size=batch_size,
+    batches_in_memory=batches_in_memory,
+    target="b2",
+    set_seed=seed,
+    load_eager=True,  # Must be enabled for resampling
```
Comment (siliataider) [inline tutorials/machine_learning/ml_dataloader_resampling.py]: You can remove the comment from this line, since it's actually relevant line 77

```suggestion
    load_eager=True,
```

File: tutorials/machine_learning/ml_dataloader_resampling.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+            loss = loss_fn(model(X), y)
+        losses.append(loss.item())
+    print(f"Validation Loss: {sum(losses) / len(losses)}")
+
+
+# First, let's try to create a dataloader without resampling and see how it handles the underrepresented class.
+dl = ROOT.Experimental.ML.RDataLoader(
+    [df_major, df_minor],
+    batch_size=batch_size,
+    batches_in_memory=batches_in_memory,
```
Comment (siliataider) [inline tutorials/machine_learning/ml_dataloader_resampling.py]:
```suggestion
```

File: tutorials/machine_learning/ml_dataloader_resampling.py [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+print("Training without resampling:")
+train_model(basic_model, basic_optimizer, dl)
+
+# Now, let's try the same thing with oversampling
+# Strategy: more batches of the underrepresented class
+# Takes more time per epoch, but each epoch is more effective
+dl_oversampled = ROOT.Experimental.ML.RDataLoader(
+    [df_major, df_minor],
+    batch_size=batch_size,
+    batches_in_memory=batches_in_memory,
```
Comment (siliataider) [inline tutorials/machine_learning/ml_dataloader_resampling.py]:
```suggestion
```

File: tutorials/machine_learning/ml_dataloader_resampling.py [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+            optimizer.zero_grad()
+            loss = loss_fn(model(X), y)
+            loss.backward()
+            optimizer.step()
+    losses = []
+    for X, y in val.as_torch():
+        with torch.no_grad():
+            loss = loss_fn(model(X), y)
+        losses.append(loss.item())
+    print(f"Validation Loss: {sum(losses) / len(losses)}")
```
Comment (siliataider) [inline tutorials/machine_learning/ml_dataloader_resampling.py]: Can we print both loss and accuracy, both for training and validation?

**Review by siliataider — CHANGES_REQUESTED** [review]
Excellent @JazzyJonah ! This is a very nice tutorial to showcase the resampling functionality to users
Just some minor changes before we're good to go

**Review by siliataider — APPROVED** [review]
LGTM! Good to go once the CI is green
